sdk/go/common/resource/plugin/langruntime_plugin.go

```diff
@@ -684,13 +687,26 @@ func (h *langhost) RunPlugin(info RunPluginInfo) (io.Reader, io.Reader, context.
 	outr, outw := io.Pipe()
 	errr, errw := io.Pipe()
 
+	cts := &promise.CompletionSource[int32]{}
+
 	go func() {
 		for {
 			logging.V(10).Infoln("Waiting for plugin message")
 			msg, err := resp.Recv()
 			if err != nil {
-				contract.IgnoreError(outw.CloseWithError(err))
-				contract.IgnoreError(errw.CloseWithError(err))
+				// If there was an error receiving then signal that the plugin has exited.
+				// If err is just EOF then the plugin has exited normally, and we can exitcode 0
+				err1 := outw.Close()
+				err2 := errw.Close()
+				if errors.Is(err, io.EOF) {
+					cts.Fulfill(0)
+				} else {
+					if err1 != nil || err2 != nil {
```

Comment:
I don't think we need this condition at all.  `nil` error values are always discarded according to the docs.

Reply:
I'll add a comment but we need it because `Join(err, nil, nil) != err` it wraps it another layer, and then our "is this a grpc error" logic doesn't kick in.

Reply:
Ah great, yeah a comment explaining that would be helpful here I think.